### PR TITLE
Allow configuration of scratch org user email address

### PR DESCRIPTION
**Pull request by davidmreed**
# Critical Changes

# Changes

- Allow configuration of the email address assigned to scratch org users, with the order of priority being (1) any `adminEmail` key in the scratch org definition; (2) the `email_address` property on the scratch org configuration in `cumulusci.yml`; (3) the `user.email` configuration property in Git.

# Issues Closed

### CODE REVIEW

File: cumulusci/core/config/ScratchOrgConfig.py [hunk 1]
```diff
@@ -136,6 +136,27 @@ def password(self):
             password = self.scratch_info["password"]
         return password
 
+    @property
+    def email_address(self):
+        email_address = self.config.get("email_address")
+        if not email_address:
+            # Obtain a default email address from Git.
```
Comment (davisagli) [inline cumulusci/core/config/ScratchOrgConfig.py]: How about we factor this out into a utility function in cumulusci.utils that takes the name of the desired git config key as an argument. I suspect this isn't the only place we'll want to do something like that.

File: cumulusci/core/config/ScratchOrgConfig.py [hunk 2]
```diff
@@ -136,6 +136,27 @@ def password(self):
             password = self.scratch_info["password"]
         return password
 
+    @property
+    def email_address(self):
+        email_address = self.config.get("email_address")
```
Comment (davisagli) [inline cumulusci/core/config/ScratchOrgConfig.py]: No strong opinion but maybe we should call it `admin_email` to better match the name in scratch org definition files.

File: cumulusci/core/config/ScratchOrgConfig.py (hunk 1 above)
Comment (davidmreed) [reply cumulusci/core/config/ScratchOrgConfig.py]: 👍 Will do.

File: cumulusci/core/config/ScratchOrgConfig.py (hunk 2 above)
Comment (davidmreed) [reply cumulusci/core/config/ScratchOrgConfig.py]: I find the SFDX-introduced nomenclature rather counterintuitive. On the theory that most of the time the value will come from Git, and secondly from `cumulusci.yml`, I'd rather call it what the user would understand it to be ("my user's email address").

**Pull request merged**